src/function/builtin.rs

```diff
@@ -53,6 +70,60 @@ pub fn builtin_function(identifier: &str) -> Option<Function> {
                 }
             }),
         )),
+
+        // string functions
+
+        "downcase" => Some(Function::new(
+            Some(1),
+            Box::new(|arguments| {
+                let subject = expect_string(&arguments[0])?;
+                Ok(Value::from(subject.to_lowercase()))
+            }),
+        )),
+        "len" => Some(Function::new(
+            Some(1),
+            Box::new(|arguments| {
+                let subject = expect_string(&arguments[0])?;
+                Ok(Value::from(subject.len() as i64))
+            }),
+        )),
+        "match" => Some(Function::new(
+            Some(2),
+            Box::new(|arguments| {
+                let subject = expect_string(&arguments[0])?;
+                let re_str = expect_string(&arguments[1])?;
+                match regex_with_local_errors(re_str) {
+                    Ok(re) => Ok(Value::Boolean(re.is_match(subject))),
+                    Err(err) => Err(err)
+                }
+            }),
+        )),
+        "replace" => Some(Function::new(
```

Comment:
In Rust, the `replace` function on strings does not work with regexes I think? So I would then rather call this function `str::regex_replace` or `str::replace_matches`, to emphasize that it expects a regex.

Reply:
As per below, `regex_matches` and `regex_replace`.